zha/application/platforms/light/__init__.py

```diff
@@ -1064,6 +1093,18 @@ def _assume_group_state(self, update_params) -> None:
 
             self.maybe_emit_state_changed_event()
 
+    def restore_external_state_attributes(
+        self,
+        *,
+        state: bool | None,
+        off_with_transition: bool | None,
+        off_brightness: int | None,
+    ) -> None:
+        """Restore extra state attributes that are stored outside of the ZCL cache."""
+        self._state = state
```

Comment:
Furthermore, I'd also consider restoring `brightness`, `color_temp`, `xy_color`, (`hs_color`, ~~`color_mode`~~) for now.

I'd prefer a different solution too, especially since it's only really needed for the older Hue lights to not break if HA is restarted before they had a chance to be polled after a color change, but seeing as [Core ZHA currently does it](https://github.com/home-assistant/core/blob/61f1c8d9631bab1a4236eee405c5b0b03e8be22e/homeassistant/components/zha/light.py#L846-L865) and we have the "framework" for it now, maybe also restore that?